flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobUtils.java

```diff
@@ -64,15 +64,19 @@
 	 * @return the storage directory used by a BLOB service
 	 */
 	static File initStorageDirectory(String storageDirectory) {
-		File baseDir = storageDirectory != null ?
-				new File(storageDirectory) :
-				new File(System.getProperty("java.io.tmpdir"));
+		File baseDir;
+		if (storageDirectory == null || storageDirectory.isEmpty()) {
```

Comment:
can you change `storageDirectory.isEmpty()` to `storageDirectory.trim().isEmpty()` to remove all whitespaces before the empty-check?